lldb/test/API/python_api/type/TestTypeList.py

```diff
@@ -119,6 +119,38 @@ def test(self):
 
         self.assertEqual(task_type, task_head_pointee_type)
 
+        # Check whether we can find a directly nested type by name
+        name_type = task_type.FindDirectNestedType("name")
+        self.assertTrue(name_type)
+        self.DebugSBType(name_type)
+
+        enum_type = task_type.FindDirectNestedType("E")
+        self.assertTrue(enum_type)
+        self.DebugSBType(enum_type)
+
+        union_type = task_type.FindDirectNestedType("U")
+        self.assertTrue(union_type)
+        self.DebugSBType(union_type)
+
+        # Check that we don't find indirectly nested types
+
+        self.assertTrue(enum_type.size == 1)
```

Comment:
Shouldn't we also check that the type we found isn't the one that's inside `struct name`?

Reply:
At the moment we don't have an API to get (direct) `DeclContext` of a type, as far as I know. So I made enums distinct by using underlying type of different size.

Reply:
ah makes sense